### PR TITLE
airbyte-ci: Add basic segment tracking

### LINKED ISSUE #31255: 📊 Add segment integration and dashboard to track `airbyte-ci` usage locally

### PR DESCRIPTION
**Pull request by bnchrch**
## Problem
We dont know what amount of usage `airbyte-ci` is receiving

## Solution
Add basic segment tracking
 closes #31255

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/pipelines/telemetry.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import getpass
+import os
+import sys
+
+import segment.analytics as analytics
+
+analytics.write_key = "G6G7whgro81g9xM00kN2buclGKvcOjFd"
```
Comment (evantahler) [inline airbyte-ci/connectors/pipelines/pipelines/telemetry.py:7]: We cool checking this in?

File: airbyte-ci/connectors/pipelines/pipelines/telemetry.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    """
+    Decorator to track CLI commands with segment.io
+    """
+
+    def wrapper(*args, **kwargs):
+        ctx = args[0]
+        top_level_command = ctx.command_path
+        full_cmd = " ".join(sys.argv)
+
+        sys_user_name = getpass.getuser()
```
Comment (evantahler) [inline airbyte-ci/connectors/pipelines/pipelines/telemetry.py:26]: Any PII worries here?  I guess this is an internal tool for now

File: airbyte-ci/connectors/pipelines/pipelines/telemetry.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import segment.analytics as analytics
+
+analytics.write_key = "G6G7whgro81g9xM00kN2buclGKvcOjFd"
+analytics.send = True
+analytics.debug = True
+
+
+def _is_airbyte_user():
+    return os.getenv("AIRBYTE_ROLE") == "airbyter"
```
Comment (evantahler) [inline airbyte-ci/connectors/pipelines/pipelines/telemetry.py:13]: Nice.

File: airbyte-ci/connectors/pipelines/pipelines/telemetry.py (hunk 1 above)
Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/telemetry.py:7]: I imagine so. We already have checked in the write key for octavia cli. I dont see the difference here.

Also not too disimilar to how sentry/segment FE libraries work

**Review by evantahler — APPROVED** [review]

File: airbyte-ci/connectors/pipelines/pipelines/telemetry.py (hunk 2 above)
Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/telemetry.py:26]: Good callout. @evantahler Ill go ahead and hash their machine info

**Review by evantahler — APPROVED** [review]

**Pull request merged**